x-pack/plugins/apm/server/routes/infrastructure/route.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import * as t from 'io-ts';
+import { createApmServerRoute } from '../apm_routes/create_apm_server_route';
+import { setupRequest } from '../../lib/helpers/setup_request';
+import { environmentRt, kueryRt, rangeRt } from '../default_api_types';
+import { getInfrastructureData } from './get_infrastructure_data';
+import { getMetricIndices } from '../../lib/helpers/get_metric_indices';
+import { getHostNames } from './get_host_names';
+
+const infrastructureRoute = createApmServerRoute({
+  endpoint: 'GET /internal/apm/services/{serviceName}/infra',
```

Comment:
Can you add an API test for this route?

Reply:
> I'll try to add the API test asap, but we agreed with @dannycroft that if I don't have time, we can add them by the test plan.

Sgtm

> About the renaming of the endpoint, which one are you suggesting renaming, and which can be a good name for it?

I don't know enough about the two endpoints to have any good suggestions. It would be great if you could look at the implementation of the existing endpoint and figure out what it actually does. 
I (very) quickly skimmed it, and it looks like it queries APM indices to get host names to be used for the log integration. 
This is different from your new implementation which queries metric indices to be used for the infrastructure metrics integration.

Perhaps some of this can be part of the route name.

Reply:
Let's take a look together tomorrow, if you have time, and I can brief you about the work we have been doing.